src/components/TableComponent.vue

```diff
@@ -87,13 +110,103 @@ const dragEnd = () => {
         window.removeEventListener("mousemove", changeWidth);
     }
 };
+
+const onRowCheckBoxChange = (evt: Event, idx: number) => {
+    //@ts-ignore
+    const value = evt.target.checked;
+    let allChecked = true;
+    let allNotCheked = true;
+
+    const checked: any[] = [];
+
+    data.value = data.value.map((item, index) => {
+        if (index == idx) {
+            item.__checked = value;
+        }
+        if (!item.__checked) {
+            allChecked = false;
+        } else {
+            allNotCheked = false;
+            if (props.checkboxIndex) {
+                checked.push(item[props.checkboxIndex]);
+            }
+        }
+
+        return item;
+    });
+
+    if (allChecked) {
+        checkBoxState.value = true;
+    } else if (allNotCheked) {
+        checkBoxState.value = false;
+    } else {
+        checkBoxState.value = "indeterminate";
+    }
+    changedFromHere = true;
+    checkedValues.value = checked;
+    emit("update:checked", checked);
+};
+
+const bulkCheckChange = (evt: Event) => {
+    //@ts-ignore
+    const value = evt.target.checked;
+    const checked: any[] = [];
+    checkBoxState.value = value;
+    data.value = data.value.map((item) => {
+        item.__checked = value;
+        if (props.checkboxIndex) {
+            checked.push(item[props.checkboxIndex]);
+        }
+        return item;
+    });
+
+    changedFromHere = true;
+    checkedValues.value = checked;
+    emit("update:checked", checked);
+};
+
```

Comment:
I'm really not a fan of this part: the `checkedFromParent` function. Even though this could be useful and is quite straightforward to use, something like `tableRef.value.rows[3].check()` would be a better API for this.

I think we should just drop support for changing the checkbox values from the parent for now.

Reply:
Okay, dropping the support from changing the checkbox values from the parent for now.

Reply:
This function would still be required just by different name (maybe `initialChecked`) to mark the state as `__checked`. But if you want to omit the use of `__checked` altogether then the operations would be much messy.

Reply:
> Okay, dropping the support from changing the checkbox values from the parent for now.

Only the `watch` block would be removed.